lib/improver/standardise.py

```diff
@@ -194,10 +194,15 @@ def _regrid_to_target(self, cube, target_grid, regridded_title):
         if self.REGRID_REQUIRES_LANDMASK[self.regrid_mode]:
             cube = self._adjust_landsea(cube, target_grid)
 
-        attributes_to_inherit = (
-            {key: val for (key, val) in target_grid.attributes.items()
-             if key in MOSG_GRID_ATTRIBUTES})
-        amend_attributes(cube, attributes_to_inherit)
+        # identify grid-describing attributes on source cube that need updating
+        required_grid_attributes = [attr for attr in cube.attributes
+                                    if attr in MOSG_GRID_ATTRIBUTES]
+        # update attributes if available on target grid, otherwise remove
+        val = (lambda k, t: t.attributes[k]
+               if k in t.attributes else "remove")  # noqa: E731
+        grid_attribute_changes = {key: val(key, target_grid)
+                                  for key in required_grid_attributes}
+        amend_attributes(cube, grid_attribute_changes)
```

Comment:
This works fine, but I do wonder whether this would be clearer if we weren't using the amend_attributes function. By using this function, you have to set up the dictionary in the correct form. Using a dictionary works well if you are passing in this information from a CLI interface, but maybe it isn't the clearest way to do it here.
You could do something like:
```
        for key in required_grid_attributes:
            if key in target_grid.attributes:
                cube.attributes.update({key: target_grid.attributes[key]})
            else:
                cube.attributes.pop(key)
```

I think this may make it a bit clearer what is going on. However if you disagree I'm happy for you to leave it as it is.

Reply:
Good idea - have changed as you suggest.